### DISCUSSION TITLE
Wiki: Anlagenkonfigurationen / Plant Configurations

### DISCUSSION THREAD

**Discussion by Philip-Wiege** [open3e/open3e#158] [Wiki]
Hallo zusammen,

nur so eine Idee von mir: Macht es Sinn, dass wir aufschreiben, wer welche Anlagenkonfiguration besitzt? Natürlich freiwillig! Jeder sollte selber eintragen und entscheiden. Ich glaube so bekommen wir einen Überblick, welche Geräte wir so "im Bestand" haben. Vielleicht können wir so ja gezielt nette User fragen, ob sie etwas testen können.

Grüße
Philip

https://github.com/open3e/open3e/wiki/200-%E2%80%90-Anlagenkonfigurationen-der-Unterstützer

**Comment by philippoo66** [comment]
das musst du natürlich hübsch bewerben, sonst wird kaum jemand auch nur drüber stolpern.

für meinen Geschmack wären solche Sachen besser etwas weiter hinten eingeordnet. Bis 100 geht es eigentlich mehr um Dokumentation, Installation/Inbetreibnahme und Anbindung an Haus-Automatisierung!?

**Comment by Philip-Wiege** [comment]
Kannst du gerne verschieben! Wo würden wir denn am besten dafür werben?

**Reply by philippoo66** [reply]
im Readme, im V Thread, bei jedem geklärten Issue, ....

**Reply by philippoo66** [reply]
ich habe es mal Seite 200 getauft... https://github.com/open3e/open3e/wiki/200-%E2%80%90-Anlagenkonfigurationen-der-Unterst%C3%BCtzer

**Comment by riessal** [comment]
Ohne eine großer Unterstützer zu sein habe ich mal meine Konfiguration auch aufgeschrieben. Mit etwas Glück wird Ende dieser Woche meine Lüfungsanlage (Vitoair 300) auch über den externen CanBus integriert.

**Reply by Philip-Wiege** [reply]
Sehr cool, danke! Hast du auch vor, die VitoAir auszulesen/steuern über Open3E?

**Reply by riessal** [reply]
Danke dir! 

Wie bei der Wärmepumpe geht es mir bei VitoAir in erster Linie darum Daten per Open3E zu gewinnen und darzustellen. Die Anlage läuft erst seit 9 Monaten und ich muss erst mal die Regelungsdynamik verstehen - und brauche Zeit das zu durchdenken.

Technisches Hauptproblem ist derzeit, dass es Viessmann erst mal fertig bringen muss die WP und die Lüftung zu koppeln (Stichwort "Systemintegration"). Das muss über den "Vitoconnect V" geschehen, da die Lüftung an sich über keinen CAN-Bus Anschluss verfügt. Sollte das nicht gehen werde ich einen zweiten Adapter benötigen und muss diesen dann mit der Lüftungssteuerung separat verbinden.

Zum Thema Steuern: 
Meine PV Anlage (Fronius) wird demnächst mit einem Batteriespeicher versorgt, im gleichen Zug will ich die SmartGrid Funktionen nutzen und per HomeAssistant und Shelly schalten (ähnlich zum Projekt im Wiki). Aber ich bin natürlich offen für Steuerung via CAN-Bus, da sind die Grenzen der Regelung ja nicht auf vier Zustände begrenzt!

Beste Grüße,

Alex

**Reply by DpunktS** [reply]
@riessal was für eine Vitoair hast du? Meine Vitoair FS 300E hat einen CAN Bus und lässt sich direkt integrieren.

**Reply by riessal** [reply]
Ich habe eine "Vitovent 300-W". Diese hat keinen CAN-Bus sondern wird laut Viessmann über einen "Vitoconnect V" erst OneBase fähig gemacht. Im "Vitoconnect V" gibt es einen CAN-Bus Anschluss.

**Reply by philippoo66** [reply]
hi @riessal !

~~bis das mit der "Vitoconnect V - Systemintegration" geklappt hat, ist für dich vielleicht mein Projekt Optolink-Splitter interessant: https://github.com/philippoo66/optolink-splitter In den Discussions und in der Wiki findest du auch viel zu Datenpunkten. Feel free eine neue Diskussion zur Vitovent aufzumachen. Der Splitter behindert eine mögliche OneBase Integration nicht, weil er die Vitoconnnect-Daten einfach nur durchleitet.~~

Vergiss es. Wenn das Vitoconnect V einen CAN Bus Anschluss hat, sollte das mit open3e doch alles kein Problem sein, oder? 

Grüsse!
Phil

**Reply by philippoo66** [reply]
~~@riessal  ich hab eine [Bedienungsanleitung](https://www.loebbeshop.de/media/67686/file/static/pdf/viessmann/bedienungsanleitung-vitoconnect-v.pdf) zu dem Vitoconnect V gefunden. Da ist nix von einem CAN Bus zu erkennen oder zu lesen. Die 'Integration' in's OneBase wird 'in der Viessmann Cloud' erfolgen. Damit hättest du dann vermutlich die gleichen Datenpunkte wie bei einer Vitoair mit CAN, aber reichlich verzögert und ggf. auch be-/eingeschränkt.~~

**Reply by philippoo66** [reply]
oh, in der Serviceanleitung gibt es CAN Bus am Vitoconnect-V ('L', Stecker 91)

![grafik](https://github.com/user-attachments/assets/eaa83852-8752-42ae-b2ce-08ac551f960d)

Wenn es vernünftig gemacht ist, sollte sich damit deine Vitoair onhe irgendwelches weiteres Zutun genau so verhalten wie die FS von DpunktS. Aber ob 'vernünftig' auf Systemintegration anzuwenden ist, wenn Viessmann ihre Finger imm Spiel haben?? (das eigentliche OneBase System ist ja zugekauft...)

**Reply by riessal** [reply]
> hi @riessal !
> 
> 
> 
> ~~bis das mit der "Vitoconnect V - Systemintegration" geklappt hat, ist für dich vielleicht mein Projekt Optolink-Splitter interessant: https://github.com/philippoo66/optolink-splitter In den Discussions und in der Wiki findest du auch viel zu Datenpunkten. Feel free eine neue Diskussion zur Vitovent aufzumachen. Der Splitter behindert eine mögliche OneBase Integration nicht, weil er die Vitoconnnect-Daten einfach nur durchleitet.~~
> 
> 
> 
> Vergiss es. Wenn das Vitoconnect V einen CAN Bus Anschluss hat, sollte das mit open3e doch alles kein Problem sein, oder? 
> 
> 
> 
> Grüsse!
> 
> Phil
> 
> 

Hey Phil,

Davon gehe ich aus. Bisher war eher das Problem dass Viessmann die Integration der beiden Geräte nicht hinbekommen hat - bzw. der Techniker der a war. Insofern bin ich gespannt auf das was am Freitag passiert. Denn wenn das klappt muss ich erst mal a die DIDs der Lüftung kommen - ein spannendes Unterfangen. Und ich muss meinen CAN Raspi endlich so konfigurieren, dass er bei einem Neustart Open3E laufen lässt - ich bin zwar extrem Technikaffin, meine Zeit aber limitiert. Daher hab ich das erst mal ans Ende gestellt.
Danke für deine Hilfe!
Alex

**Reply by riessal** [reply]
> ~~@riessal  ich hab eine [Bedienungsanleitung](https://www.loebbeshop.de/media/67686/file/static/pdf/viessmann/bedienungsanleitung-vitoconnect-v.pdf) zu dem Vitoconnect V gefunden. Da ist nix von einem CAN Bus zu erkennen oder zu lesen. Die 'Integration' in's OneBase wird 'in der Viessmann Cloud' erfolgen. Damit hättest du dann vermutlich die gleichen Datenpunkte wie bei einer Vitoair mit CAN, aber reichlich verzögert und ggf. auch be-/eingeschränkt.~~ 

Problem ist wohl dass die Vitoair auf der Onebase Platform entwickelt wurde und somit einen CAN Bus hat, die Vitovent leider nicht. Dafür Modbus. Über diesen wird der Vitoconnect angeschlossen der dann entweder per WLAN an Viessmann überträgt oder per CAN Bus a die Vitocal angeschlossen.

**Reply by philippoo66** [reply]
oh, ok, jetzt hatte ich Vitoair und Vitovent in einen Topf geworfen.

> wenn das klappt muss ich erst mal a die DIDs der Lüftung kommen - ein spannendes Unterfangen.

das wiederum sollte dann ganz einfach sein: einfach nur depictSystem laufen lassen! Das sucht die vorhandenen Datenpunkte zusammen und schreibt sie schön nach ECUs geordnet in die datapoints_6xy.py Dateien

**Reply by riessal** [reply]
Kurzes Update: der Viessmann Techniker ist erkrankt. Nächster Termin Ende Januar. *thisisfine*

**Reply by philippoo66** [reply]
das ist immer toll wenn man im Winter 6 Wochen auf nen Heizungsmensch warten muss.............................

**Reply by riessal** [reply]
Well, funktioniert alles, nur nicht integriert. Und darum geht's ja. Viel mehr nervt die Tatsache, dass man Updates hinterherlaufen muss...

**Reply by riessal** [reply]
Good news everyone. Ein sehr netter Techniker war bei mir, hat die Lüftung via externem CAN-Bus mit der Wärmepumpe verbunden und sogar Verständnis dafür gehabt, dass ich weiter lokal Daten will. Insofern war er so nett und hat die Lüftung innen angeschlossen sodass ich meinen CAN-Bus Reader weiter einfach unten anstecken kann. Jetzt bin ich gespannt was ich zusätzlich sehe! ;-)

**Reply by philippoo66** [reply]
und @riessal ?! schon ein neues depictSystem gemacht?

**Reply by riessal** [reply]
Ja, sorry für die späte Antwort. 
War leider recht ernüchternd, ich kann dir das Ergebnis im Verlauf zukommen lassen. Viel neues habe ich nicht gefunden. 
Die Lüftung hängt jetzt per Modbus am Vitoconnect V und der per CAN-Bus an der Vitocal. Damit kann ich jetzt kein Steuergerät mit Display mehr anschließen. Das liegt jetzt abgesteckt auf der Lüftung und muss auch da bleiben, denn wenn mal ein internes Teil in der Lüftung getauscht wird muss damit Konfiguriert werden und dann wieder der Vitoconnect V ran. :-(

**Reply by riessal** [reply]
Lach nicht, jetzt ist es integriert. Kann ich dir die Dateien nicht-öffentlich schicken?

**Reply by philippoo66** [reply]
klar. open3e@web.de 
Warum sollte ich lachen?

> Damit kann ich jetzt kein Steuergerät mit Display mehr anschließen

warum? ist das per Modbus? CAN ist doch ein multi-master System, da kann man so einiges 'parallel' dran anschliessen...

**Reply by riessal** [reply]
> klar. [open3e@web.de](mailto:open3e@web.de) Warum sollte ich lachen?

Weil es so lange gedauert hat! :-)

> 
> > Damit kann ich jetzt kein Steuergerät mit Display mehr anschließen
> 
> warum? ist das per Modbus? CAN ist doch ein multi-master System, da kann man so einiges 'parallel' dran anschliessen...

Das lässt der Vitoconnect V scheinbar nicht zu. Und die Vitovent auch nicht. Entweder/Oder. Jetzt liegt das Display quasi tot auf der Lüftung...falls mal was is....

Mail ist unterwegs! VG

**Reply by riessal** [reply]
Angekommen?

**Reply by philippoo66** [reply]
jap, is da, danke! musste erstma aus dem Unbekannt Filter raus...

**Reply by riessal** [reply]
Man möchte es kaum glauben, aber heute war mal wieder einer der Viessmänner da und jetzt ist alles integriert. Ich würde dir nochmal mein System schicken?

**Reply by philippoo66** [reply]
gerne! aber wie gesagt - optimal wäre depictSystem mit Option -s und dann die devices.json, die datapoints_6yz.py _und_ die Virtdata_6yz.txt. Dann kann ich es in die Simulation laden :-)

**Reply by riessal** [reply]
Hab dir eine weitere Mail geschickt :-)

**Reply by philippoo66** [reply]
angekommen, danke! :-)

**Reply by riessal** [reply]
Hat sich was sinnvolles ergeben? VG

**Comment by wutz-ESP** [comment]
Wie kann man mit den Benutzern der 200 ‐ Anlagenkonfigurationen der Unterstützer in Kontakt kommen?
Habe eine Vitodens 300-W B3HG-25 und würde gerne open3e nutzen um Betriebsdaten in fhem zu bekommen.
@Dreman65 hat eine B3HG - bei mir sind es 2 Heizkreise und externe Ansteuerung von 2 WW Speichern.

**Reply by Dreman65** [reply]
@wutz-ESP  Woran hakt es denn konkret? Prinzipiell ist die Inbetriebnahme von open3e oder e3oncan zumindest bei den Vitodensen unabhängig von der Vitodens-Konfiguration. Diese Konfigurationssammlung hier hat ja mehr das Ziel  '_vielleicht können wir so ja gezielt nette User fragen, ob sie etwas testen können.'_

Und wenn es dann (trotz sehr detaillierter Readme hier) bei der Inbetriebnahme klemmt: entweder hier unter 'Discussions' nachfragen oder im [Viessmann-Forum](https://community.viessmann.de/t5/Konnektivitaet/CAN-Bus-Home-Automation-E3-Generation-lokal-und-kostenlos/td-p/356066) nachfragen.

Inwieweit es bereits User mit Erfahrung im Zusammenspiel von open3e und fhem gibt: keine Ahnung :-).

**Comment by wutz-ESP** [comment]
@Dreman65 Hab inzwischen den CAN-Adapter wie beschrieben in Nutzung und frage auf einem Testsystem die Werte alle 60 Sekunden ab.

Mein Problem ist jetzt die Bezeichnung der Heizkreise

08:49:11.671 | 334 | open3e/627_CentralHeatingOneCircuitName | "**4865697a6b6f657270657200**"
da steht in der ViCare-App  **Heizkoerper**

08:49:11.721 | 334 | open3e/628_CentralHeatingTwoCircuitName | "**46757373626f64656e20487a672e00**"
da steht in der ViCare-App  **Fussboden Hzg.**

Habe schon versucht in der Open3Edatapoints_680.py die Einträge

        627 : RawCodec(12, "CentralHeatingOneCircuitName"),
        628 : RawCodec(15, "CentralHeatingTwoCircuitName"),
        629 : RawCodec(12, "CentralHeatingThreeCircuitName"),
        630 : RawCodec(12, "CentralHeatingFourCircuitName"),

auf  627 : None,   zu ändern dann kommt aber eine Fehlermeldung

**Reply by Dreman65** [reply]
@wutz-ESP Ändere die RawCodec-Zeilen mal in

```
627 : O3EUtf8(12, "CentralHeatingOneCircuitName"), 
628 : O3EUtf8(12, "CentralHeatingTwoCircuitName"),
629 : O3EUtf8(12, "CentralHeatingThreeCircuitName"),
630 : O3EUtf8(12, "CentralHeatingFourCircuitName"),
```
dann sollte Lesen und Schreiben funktionieren.

**Comment by wutz-ESP** [comment]
Danke mit 
627 : O3EUtf8(12, "CentralHeatingOneCircuitName"), 
628 : O3EUtf8(**15**, "CentralHeatingTwoCircuitName"),
funktioniert es

**Reply by Dreman65** [reply]
Warum bei dir die Bytelänge bei Parameter 628 15 Bytes ist. keine Ahnung, bei meiner 300 W sind die Parameter 627 bis 634 einheitlich 12 Byte lang. Ist halt Viessmann :-).
